### DISCUSSION TITLE
Pregunta sobre los forks

### DISCUSSION THREAD

**Discussion by venegas55** [LaunchX-InnovaccionVirtual/CursoIntroPython#153] [Q&A] [answered]
Creo que la Commander Fer ya lo había comentado el miércoles, pero....

si yo tengo este repositorio "forkeado" en mi perfil y hago commits con los cambios que hago en las katas.... ¿Tanto la commander Fer como la Organización pueden ver esos commits? o solo hasta que haga un pull request...

Python no me está costando problema pero aunque las 2 clases de git fueron buenas... siento que me falta un MADRAL por aprender jejeje

salu2!

**Comment by arturodelac** [comment]
Primero tendrías que hacer un pull request y fer tendría que aceptarlo

**Comment by oscar-cbrlc** [comment]
El fork es como "copiar" un repositorio. Entonces, si haces commits, esos cambios se van a hacer en esa copia tuya. El repositorio original no se verá afectado por eso. El pull request es para que se hagan los cambios en el repo original, pero como ya sabrás... no debes hacer eso en este repo

**Comment by oscardanielM** [comment]
Hola, si Fer tendría que aceptar tu cambio, pero comento que seria mejor que  tu armes tu repositorio en tu perfil y solo descargar el proyecto en local y después subir tus cabios a tu repositorio en git.

Así tendrías tu repositorio y como ella menciono el miércoles en el live, solo compartirías el link de tu proyecto y otras cosas en el form que nos estaria enviando.

 😁👍

**Reply by RodolfoMorquecho** [reply]
Hola Oscar, espero me puedas ayudar con esta duda. Ya tengo mis katas terminadas pero no he hecho eso del fork ni el repositorio porque hay un enlace que dice que hoy se dira como se subiran, Tenia que haber ido subiendo una por una en el repositorio?

**Reply by JLuisPrz** [reply]
Hola Rodo, te recomiendo que hagas tu propio repositorio y subas todas tus katas en ese repositorio. Así vas practicando y le vas entendiendo más.
En este enlace te explica como subir tu proyecto a GitHub.
https://intelequia.com/blog/post/2121/c%C3%B3mo-subir-un-proyecto-a-un-repositorio-de-github
Si tienes alguna otra duda, sin problema te puedo ayudar.

**Reply by RodolfoMorquecho** [reply]
Gracias Luis por tu respuesta me ayuda mucho, le dare una leída para comprenderlo. Otra pregunta al subirlo a el repositorio como lo explica el link que me pusiste es como lo pedira la profesora para entregarle las katas o esa es otra cosa?

**Reply by JLuisPrz** [reply]
Por lo que entendí, solo tenemos que mandar el link de nuestro repositorio y nuestro nombre. Pero ahorita ya lo va explicar en el live.

**Reply by RodolfoMorquecho** [reply]
Si ya vi que todo se va a mandar en un link, la duda que yo tenia es que como todas mis katas las realice de forma manual/local sin descargar el zip de archivos no había hecho fork ni el repositorio y creo que lo del fork no es necesario así.

**Reply by JLuisPrz** [reply]
No, el fork no es necesario. Haz tu propio repo y sube todas tus katas.

**Comment by itsAmauriDev** [comment]
Hola!
Como ya te comentaron no hay ningún inconveniente ya que el fork crea una copia, no trabajas en el repositorio original, solo recuerda que si vas a estar trabajando con ese repositorio tienes que estar actualizando los cambios que se hagan en repositorio original(el de Fer) para tener la versión más reciente.

Si quieres conocer un poco más  acerca del funcionamiento de git, puedes ver el curso gratuito de Código Facilito. Con las primeras 5 clases sera suficiente para que entiendas las bases.
[https://codigofacilito.com/cursos/git](https://codigofacilito.com/cursos/git)

**Comment by doguedogue** [comment]
En mi opinión
- Para este caso no es como un repositorio colaborativo para hacer fork, 
- lo ideal es hacer clonarlo /descargarlo para usar los notebooks en tu local
- posteriormente trabajar con otro repositorio para tus entregables

**Comment by DanielOe21** [comment, marked as answer]
# Fork vs Clone

Ambas son formas para "copiar" un repositorio, pero una forma sencilla para entenderlo sería:

- Fork es una copia que sigue estando ligada al repositorio original. Si la original sufre cambios, la copia la puedes actualizar (manualmente). Puedes hacer cambios en tu Fork y hacer una solicitud para integrar los cambios al original.

- Clon es una copia exacta del repositorio en ese preciso momento en el que lo hagas, pero ya no está ligada al original, puedes hacer con él lo que quieras sin afectar al otro.

**Reply by NayGL** [reply]
Gracias por la explicación :)

**Reply by venegas55** [reply]
claro, entiendo esa parte. Clonar sería en todo caso bajar el zip y tenerlo local y ya luego subirlo a una repo propia....

me interesa actualizar el fork con la original... segun yo no se le han hecho cambios desde que hice fork... pero me interesaría ver si se le pueden añadir.

Trabajé mis katas de manera "local" por que vinculé github con visual studio, tons ahí abro la repo, hago cambios y los hago commit a mi repo "forkeada"
